functionary/ui/views/tasks.py

```diff
@@ -188,3 +199,30 @@ def get(self, request, *args, **kwargs):
             )
 
         return render(request, "partials/task_result.html", context=context)
+
+
+@require_GET
+@login_required
+def get_task_log(request: HttpRequest) -> HttpResponse:
+    env = Environment.objects.get(id=request.session.get("environment_id"))
+    if not request.user.has_perm(Permission.TASK_READ, env):
+        return HttpResponseForbidden()
+
+    if not (task_id := request.GET.get("task_id")):
+        return HttpResponse("Unknown task submitted.")
+
+    task = None
+    try:
+        # Use try/except in case of invalid task_id format
+        task = Task.objects.get(id=task_id, environment=env)
```

Comment:
There's a `get_object_or_404` helper in django.shortcuts or some such thing.  It handles this try except and response for you.  You may or may not still have to handle the `ValidationError` for malformed uuids.

Reply:
I confirmed that you do still need to handle the ValidationError for malformed UUID strings